src/components/inventory/Inventory.tsx

```diff
@@ -1,27 +1,38 @@
 'use client';
 
 import { useEffect, useState } from 'react';
-import { useSources } from '../../hooks/sources/useSources';
+import { useSources, useUpdateSources } from '../../hooks/sources/useSources';
 import { useSetSourceToPurge } from '../../hooks/sources/useSetSourceToPurge';
-import { SourceWithId } from '../../interfaces/Source';
+import { SourceWithId, SrtSource } from '../../interfaces/Source';
 import EditView from './editView/EditView';
 import SourceList from '../sourceList/SourceList';
 import { useTranslate } from '../../i18n/useTranslate';
 import { useRemoveInventorySourceItem } from '../../hooks/sources/useRemoveInventorySource';
+import { useCreateSrtSource } from '../../hooks/sources/useCreateSrtSource';
+import { AddSrtModal } from '../modal/addSrtModal/AddSrtModal';
+import { IconReload } from '@tabler/icons-react';
+import { Loader } from '../loader/Loader';
 
 export default function Inventory({ locked }: { locked: boolean }) {
   const [purgeInventorySource, reloadList] = useSetSourceToPurge();
   const [removeInventorySourceItem, reloadInventoryList] =
     useRemoveInventorySourceItem();
+  const [createSrtSource, reloadSourceList] = useCreateSrtSource();
+
   const [updatedSource, setUpdatedSource] = useState<
     SourceWithId | undefined
   >();
-  const [sources] = useSources(
-    reloadList || reloadInventoryList,
-    updatedSource
+  const [refreshKey, setRefreshKey] = useState<number>(0);
+  const [sources, loadingSources] = useSources(
+    reloadList || reloadInventoryList || reloadSourceList || !!refreshKey,
+    updatedSource,
+    refreshKey
   );
+
+  const [updateSources, updateSourcesLoading] = useUpdateSources();
   const [currentSource, setCurrentSource] = useState<SourceWithId | null>();
   const t = useTranslate();
+  const [showSrtModal, setShowSrtModal] = useState<boolean>(false);
```

Comment:
A little unstructured maybe. 
Instead of keeping it:
- states
-hook-calls
-variables